src/main/java/ch/njol/skript/conditions/CondHasCustomModelData.java

```diff
@@ -0,0 +1,55 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.conditions;
+
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+
+@Name("Has Custom Model Data")
+@Description("Check if an item has a custom model data tag")
+@Examples("player's tool has custom model data")
+@Since("INSERT VERSION")
+public class CondHasCustomModelData extends PropertyCondition<ItemStack> {
+	
+	static {
+		if (Skript.methodExists(ItemMeta.class, "hasCustomModelData")) {
+			register(CondHasCustomModelData.class, PropertyType.HAVE, "[custom] model data", "itemstacks");
```

Comment:
Should be taking ItemType here

Reply:
I had a really really hard time creating this with item type, when I first made it a few months ago in Sk-NBeeT, but I will attempt again.

Reply:
fixed